src/recon_test/test_consistency_with_root.cxx

```diff
@@ -0,0 +1,402 @@
+  {}
+
+  //! Coordinate position
+  CartesianCoordinate3D<float> coord;
+  //! Weight value
+  float value;
+};
+
+
+class ROOTConsistencyTests : public RunTests
+{
+public:
+  ROOTConsistencyTests()= default;
+
+  /*!
+   * \brief Run the tests with ROOT data
+
+   * \returns 0 if all tests passed, 1 otherwise
+   */
+  void run_tests() override;
+
+private:
+
+  void setup();
+  /*! Reads listmode event by event, computes the ProjMatrixElemsForOneBin (probabilities
+   * along a bin LOR). This method should include all tests/ setup for additional test such that the list data is
+   * read only once.
+   *
+   * Passes ProjMatrixElemsForOneBin (LOR) to test_nonTOF_LOR_closest_approach() and if fails,
+   * add 1 to `failed_events` (LOR's closest voxel was not within nonTOF_distance_threshold).
+   * Check if the number of `failed_events` is greater than half the number of tested events to pass the test.
+   */
+  void process_list_data();
+
+  void loop_through_list_events(shared_ptr<CListModeData> lm_data_sptr, const shared_ptr<ProjMatrixByBin>& proj_matrix_sptr,
+                                const CartesianCoordinate3D<float> original_coords);
+  /*! Given a ProjMatrixElemsForOneBin (probabilities), test if the closest voxel in the LOR to the original_coords
+   * is within nonTOF_distance_threshold distance. If it is, pass with true, otherwise fales.
+   * @param probabilities ProjMatrixElemsForOneBin object of a list mode event
+   * @return True if test passes, false if failed.
+   */
+  bool test_nonTOF_LOR_closest_approach(const ProjMatrixElemsForOneBin& probabilities);
+  void post_processing_nonTOF();
+
+
+  //! Selects and stores the highest probability elements of ProjMatrixElemsForOneBin.
+  bool test_TOF_max_lor_voxel(const ProjMatrixElemsForOneBin& probabilities);
+
+  //! Checks if original and calculated coordinates are close enough and saves data to file.
+  void post_processing_TOF();
+
+  // Auxilary methods
+  std::string get_root_header_filename() { return "pretest_output/root_header_test" + std::to_string(test_index) + ".hroot"; }
+  std::string get_generate_image_par_filename() { return "SourceFiles/generate_image" + std::to_string(test_index) + ".par"; }
+
+
+  ///// Class VARIABLES /////
+
+  // Data set variables
+  int test_index;
+  int num_tests;
+
+  // Test results storage. Records if each test failed or not. E.g.`test_index` result is stored in `test_results_nonTOF[test_index -1]`
+  std::vector<bool> test_results_nonTOF;
+  std::vector<bool> test_results_TOF;
+
+  shared_ptr<DiscretisedDensity<3, float> > discretised_density_sptr;
+  CartesianCoordinate3D<float> original_coords; // Stored in class because of dynamic_cast
+  CartesianCoordinate3D<float> grid_spacing;    // Stored in class because of dynamic_cast
+
+  int num_events_tested = 0;
+
+  /// NON TOF VARIABLES
+  //! A vector to store the coordinates of the closest approach voxels.
+  const float failure_tolerance_nonTOF = 0.05;
```

Comment:
this variable needs to be moved before the doxygen comment, but it also needs its own comment

Reply:
Done